kuka_rsi_hw_interface/src/kuka_hardware_interface.cpp

```diff
@@ -234,6 +235,7 @@ return_type KukaRSIHardwareInterface::read(
     return return_type::ERROR;
   }
   if (!command_handler_.Decode(in_buffer_, UDP_BUFFER_SIZE)) {
+    this->on_deactivate(this->get_state());
```

Comment:
are these deactivations necessary? I would assume returning failure has the same effect

Reply:
I see, I encountered an error like this today.
I closed the simulator while the there was an active control and the whole driver died. It was like the control loop was running without any wait.
We should put it into an issue

Reply:
You should deactivate the manager node in case of an error, but here you deactivate the hw if, which is IMO done by simply returning error.

Reply:
I deleted it and resolved this comment